libs/vendored/jwtool/src/lib/jwtool.ts

```diff
@@ -0,0 +1,207 @@
+    if (obj.d) {
+      return new PrivateJWK(obj, pem);
+    }
+    return new PublicJWK(obj, pem);
+  }
+
+  static fromObject(obj: any, extras?: any): PublicJWK | PrivateJWK {
+    obj = addExtras(obj, extras);
+    if (obj.d) {
+      return new PrivateJWK(obj);
+    }
+    return new PublicJWK(obj);
+  }
+
+  static fromFile(filename: string, extras?: any): PublicJWK | PrivateJWK {
+    const file = fs.readFileSync(filename, 'utf8');
+    if (file[0] === '{') {
+      return JWK.fromObject(JSON.parse(file), extras);
+    }
+    return JWK.fromPEM(file, extras);
+  }
+
+  toJSON() {
+    return this.jwk;
+  }
+}
+
+export class PrivateJWK extends JWK {
+  signSync(data?: any): string {
+    const payload = data || {};
+    payload.iss = this.jwk.iss;
+    return sign(
+      {
+        header: {
+          alg: 'RS256',
+          jku: this.jwk.jku,
+          kid: this.jwk.kid,
+        },
+        payload: payload,
+      },
+      this.pem
+    );
+  }
+
+  async sign(data?: any): Promise<string> {
+    return Promise.resolve(this.signSync(data));
+  }
+}
+
+export class PublicJWK extends JWK {
+  verifySync(str: string): any {
+    return verify(str, this.pem);
+  }
+
+  async verify(str: string): Promise<any> {
+    return Promise.resolve(this.verifySync(str));
+  }
+}
+
+export class JWTVerificationError extends Error {
+  constructor(msg: string) {
+    super(msg);
+    this.name = 'JWTVerificationError';
+    this.message = msg;
+  }
+}
+
+async function getJwkSet(jku: string) {
+  try {
+    const res = await fetch(jku, {
+      redirect: 'error',
+    });
+    const body = await res.text();
+    const json = JSON.parse(body);
+    const set: Record<string, PublicJWK> = {};
+    json.keys.forEach((key: any) => {
+      set[key.kid] = new PublicJWK(key);
+    });
+    return set;
+  } catch (e) {
+    throw new JWTVerificationError('bad jku');
+  }
+}
+
+export class JWTool {
+  trusted: string[];
+  jwkSets: Record<string, Record<string, PublicJWK>>;
+
+  constructor(trusted: string[]) {
+    this.trusted = trusted;
+    this.jwkSets = {};
+  }
+
+  static JWK = JWK;
+  static PublicJWK = PublicJWK;
+  static PrivateJWK = PrivateJWK;
+  static JWTVerificationError = JWTVerificationError;
+  static unverify = decode;
+  static verify = verify;
+  static sign = sign;
+
+  async fetch(jku: string, kid: string): Promise<PublicJWK> {
+    let set = this.jwkSets[jku];
+    if (set && set[kid]) {
+      return set[kid];
+    }
+    set = await getJwkSet(jku);
+    this.jwkSets[jku] = set;
+    if (!set[kid]) {
+      throw new JWTVerificationError('unknown kid');
+    }
+    return set[kid];
+  }
+
+  async verify(str: string, defaults?: any): Promise<any> {
+    defaults = defaults || {};
+    let jwt = decode(str);
+    if (!jwt) {
+      throw new JWTVerificationError('malformed');
+    }
+
+    const jku = jwt.header.jku || defaults.jku;
+    const kid = jwt.header.kid || defaults.kid;
+    if (this.trusted.indexOf(jku) === -1) {
```

Comment:
nit: since it looks like `this.trusted` is a `string[]`, you could use [`Array#includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) which returns a Boolean (vs comparing to a negative index).
```suggestion
    if (!this.trusted.includes(jku)) {
```